app/src/main/java/io/gnosis/safe/notifications/NotificationRepository.kt

```diff
@@ -224,9 +229,24 @@ class NotificationRepository(
             }
             .onFailure {
                 registrationUpdateFailed = true
+                handleCloudMessagingTokenIsLinkedToAnotherDeviceError(it)
             }
     }
 
+    private suspend fun handleCloudMessagingTokenIsLinkedToAnotherDeviceError(throwable: Throwable) {
+        val error = throwable.toError()
+        if (error == Error.ErrorCloudMessagingTokenIsLinkedToAnotherDevice) {
+            resetFirebaseToken()
+        }
+    }
+
+    private suspend fun resetFirebaseToken() {
+        kotlin.runCatching {
+            FirebaseInstanceId.getInstance().deleteInstanceId()
```

Comment:
suggestion to make it more legible:
```kotlin
with(FirebaseInstanceId.getInstance()){
  deleteInstanceId()
  deleteToken(getCloudMessagingToken()!!, INSTANCE_ID_SCOPE)
}
```

Reply:
can we always assume token won't be null? Otherwise it will crash here

Reply:
Solved